presto-hive/src/test/java/io/prestosql/plugin/hive/AbstractTestHive.java

```diff
@@ -2539,6 +2540,59 @@ public void testEmptyTableCreation()
         }
     }
 
+    @Test
+    public void testCreateEmptyTableStaging()
+    {
+        for (HiveStorageFormat storageFormat : createTableFormats) {
+            SchemaTableName temporaryCreateEmptyTable = temporaryTable("create_empty_staging");
+            try {
+                List<Column> columns = ImmutableList.of(new Column("test", HIVE_STRING, Optional.empty()));
+                try (Transaction transaction = newTransaction()) {
+                    ConnectorSession session = newSession();
+                    String tableOwner = session.getUser();
+                    String schemaName = temporaryCreateEmptyTable.getSchemaName();
+                    String tableName = temporaryCreateEmptyTable.getTableName();
+                    LocationService locationService = getLocationService();
+                    LocationHandle locationHandle = locationService.forNewTable(transaction.getMetastore(),
+                            session,
+                            schemaName,
+                            tableName,
+                            Optional.empty(),
+                            false);
+                    Path writePath = locationService.getQueryWriteInfo(locationHandle).getWritePath();
+                    Path targetPath = locationService.getQueryWriteInfo(locationHandle).getTargetPath();
+                    Table.Builder tableBuilder = Table.builder()
+                            .setDatabaseName(schemaName)
+                            .setTableName(tableName)
+                            .setOwner(tableOwner)
+                            .setTableType(TableType.MANAGED_TABLE.name())
+                            .setParameters(ImmutableMap.of(
+                                    PRESTO_VERSION_NAME, TEST_SERVER_VERSION,
+                                    PRESTO_QUERY_ID_NAME, session.getQueryId()))
+                            .setDataColumns(columns)
+                            .setPartitionColumns(ImmutableList.of());
+                    tableBuilder.getStorageBuilder()
+                            .setLocation(targetPath.toString())
+                            .setStorageFormat(StorageFormat.create(storageFormat.getSerDe(), storageFormat.getInputFormat(), storageFormat.getOutputFormat()))
+                            .setBucketProperty(Optional.empty())
+                            .setSerdeParameters(ImmutableMap.of());
+                    PrincipalPrivileges principalPrivileges = testingPrincipalPrivilege(tableOwner, session.getUser());
+                    transaction.getMetastore().createTable(session, tableBuilder.build(), principalPrivileges, Optional.empty(), true, EMPTY_TABLE_STATISTICS);
+                    transaction.commit();
+
+                    HdfsContext context = new HdfsContext(session, schemaName, tableName);
+                    if (locationHandle.getWriteMode() == STAGE_AND_MOVE_TO_TARGET_DIRECTORY) {
+                        assertFalse(pathExists(context, hdfsEnvironment, writePath),
```

Comment:
`io.prestosql.plugin.hive.util.HiveWriteUtils#createTemporaryPath` uses `randomUUID`, so different temp directory will be used in `transaction.getMetastore().createTable` than `writePath`.

What you should do instead is to set some random staging directory (via `temporary_staging_directory_path` session property) and check that no dir was created there after `createTable`